packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -369,6 +371,30 @@ class FlutterDriverExtension {
     return null;
   }
 
+  /// Returns a future that waits until frame is synced.
+  ///
+  /// Specifically, it checks:
+  /// * Whether the count of transient callbacks is zero.
+  /// * Whether there's no pending request for scheduling a new frame.
+  /// We consider the frame is synced when both conditions are met.
+  ///
+  /// This method relies on a Flutter Driver mechanism called "frame sync",
+  /// which waits for transient animations to finish. Persistent animations will
+  /// cause this to wait forever.
+  ///
+  /// If a test needs to interact with the app, it should avoid this method and
```

Comment:
I know I wrote this line, but on re-reading it I think we should probably clarify `If a test needs to interact with the app` to say something like `If a test needs to interact with the app while animations are running`.

Reply:
Thanks for double checking! Done.